notebooks/Notebook1.md

```diff
@@ -18,17 +24,35 @@ exciting milestones, preliminary results, etc. Feel free to include images
+
+I implemented everything this week in Python, as it was quick and simple to get my ideas down, which let me find the places where a DSL could really fill in the gaps. However, I'm still confident that I want to implement the actual DSL in Scala, as I think the features it offers, both in terms of customizability and OOP in general, will allow for a better-constructed language. This is a bit of a tradeoff, however. Having a DSL hosted in Scala rather than in Python means that a prospective programmer will have a larger barrier to entry. As I alluded to in my first section, though, this will likely mean that the language is easier to read, but harder to write (at least, for someone with little CS experience). Since I consider the functionality and readability of the language to be more important than its writeability, I think this is a tradeoff worth making. While I'm a bit disappointed that the language will not be as easy to use as I had originally intended, I think this is in the language's best interest. With many more weeks of development, I feel confident that a GUI could be devised, perhaps Scratch-style, that would make development in this DSL as easy as I want it to be, without sacrificing its flexibility or readability.
+
+https://docs.google.com/document/d/1yPfdPlC9sN5qvzx74-tXSt-7NBsu8DFBAY_Z_-y3AB8/edit?usp=sharing
+
+I have actually implemented all of the features on the first page of this document in Python (which I used to construct the Hearts program), but the features on the second page are the ones currently giving me trouble.
+
 ## Questions
 
 **What is the most pressing issue for your project? What design decision do
 you need to make, what implementation issue are you trying to solve, or how
 are you evaluating your design and implementation?**
 
+Far and away the most pressing issue for my project is the Round-Turn-Action-Move sequence. I feel confident that the Round implementation won't be terrible, and most of this week was spent getting Move to be as ironed out as possible. I'm just not totally certain that Scala has the right tools to make the Turn and Action sequences quite as crisp as I want them to be, but I think I'll be able to find something good enough; I just suspect it's going to take a lot of time. On the bright side, I think I've found a pretty good design loop, which I was using this week: I wrote a full program in my temporary Python DSL, then added features to that DSL to reduce the size of my program, shrinking it quite considerably (and making it much more readable). This seems like a good way to evaluate the strength of my DSL in general, although writing programs in it that serve as functional test cases is quite daunting at times, and may take up the bulk of my time in one of the later weeks.
```

Comment:
Think about it like implementing a game or a simulation. You want to have *basically* a "run" command that prompts the user and then when the game is over, gives insight.

Reply:
I love the idea of having built in "ai" in the DSL that you play with but that might be outside the scope of this project